velox/functions/prestosql/tests/BigintEnumCastTest.cpp

```diff
@@ -0,0 +1,190 @@
+      {0, -2, std::nullopt},
+      {0, -2, std::nullopt});
+
+  // Cast INTEGER to enum type
+  testCast<int32_t, int64_t>(
+      INTEGER(),
+      BIGINT_ENUM(typeParameters),
+      {0, -2, std::nullopt},
+      {0, -2, std::nullopt});
+
+  // Cast enum type to same enum type
+  testCast<int64_t, int64_t>(
+      BIGINT_ENUM(typeParameters),
+      BIGINT_ENUM(typeParameters),
+      {0, -2, std::nullopt},
+      {0, -2, std::nullopt});
+
+  // Cast base type to enum type where the value does not exist in the enum
+  VELOX_ASSERT_THROW(
+      evaluateCast(
+          BIGINT(),
+          BIGINT_ENUM(typeParameters),
+          makeRowVector({makeNullableFlatVector<int64_t>(
+              {0, 1, std::nullopt}, BIGINT())})),
+      "No value '1' in test.enum.mood");
+
+  // Cast enum type to different enum type
+  std::string enumName2 = "someEnumType";
+  std::unordered_map<std::string, int64_t> enumMap2 = {
+      {"CURIOUS", -2}, {"HAPPY", 3}};
+  LongEnumParameter longEnumParameter2(enumName2, enumMap2);
+  auto typeParameters2 = {TypeParameter(longEnumParameter2)};
+  VELOX_ASSERT_THROW(
+      evaluateCast(
+          BIGINT_ENUM(typeParameters),
+          BIGINT_ENUM(typeParameters2),
+          makeRowVector({makeNullableFlatVector<int64_t>(
+              {0, 1, std::nullopt}, BIGINT_ENUM(typeParameters))})),
+      "Cannot cast test.enum.mood:BigintEnum(test.enum.mood{\"CURIOUS\": -2, \"HAPPY\": 0}) to someEnumType:BigintEnum(someEnumType{\"CURIOUS\": -2, \"HAPPY\": 3}).");
+
+  // Cast enum type to different enum type with same name
+  LongEnumParameter longEnumParameter3(enumName, enumMap2);
+  auto typeParameters3 = {TypeParameter(longEnumParameter3)};
+  VELOX_ASSERT_THROW(
+      evaluateCast(
+          BIGINT_ENUM(typeParameters),
+          BIGINT_ENUM(typeParameters3),
+          makeRowVector({makeNullableFlatVector<int64_t>(
+              {0, 1, std::nullopt}, BIGINT_ENUM(typeParameters))})),
+      "Cannot cast test.enum.mood:BigintEnum(test.enum.mood{\"CURIOUS\": -2, \"HAPPY\": 0}) to test.enum.mood:BigintEnum(test.enum.mood{\"CURIOUS\": -2, \"HAPPY\": 3})");
+
+  // Cast varchar type to enum type
+  VELOX_ASSERT_THROW(
+      evaluateCast(
+          VARCHAR(),
+          BIGINT_ENUM(typeParameters),
+          makeRowVector({makeNullableFlatVector<StringView>(
+              {"a"_sv, "b"_sv, std::nullopt})})),
+      "Cannot cast VARCHAR to test.enum.mood:BigintEnum(test.enum.mood{\"CURIOUS\": -2, \"HAPPY\": 0}).");
+
+  // Cast boolean type to enum type
+  VELOX_ASSERT_THROW(
+      evaluateCast(
+          BOOLEAN(),
+          BIGINT_ENUM(typeParameters),
+          makeRowVector(
+              {makeNullableFlatVector<bool>({true, false, std::nullopt})})),
+      "Cannot cast BOOLEAN to test.enum.mood:BigintEnum(test.enum.mood{\"CURIOUS\": -2, \"HAPPY\": 0}).");
+}
+
+TEST_F(BigintEnumCastTest, fromBigintEnum) {
+  auto enumName = "test.enum.mood";
+  std::unordered_map<std::string, int64_t> enumMap = {
```

Comment:
Consider creating test enum types in the ctor and store them in member variables. Then reuse across all tests.